plugins/org.eclipse.fordiac.ide.gitlab/src/org/eclipse/fordiac/ide/gitlab/management/GitLabDownloader.java

```diff
@@ -94,7 +95,15 @@ public Map<Project, List<Package>> getProjectsAndPackages() {
 		return projectAndPackageMap;
 	}
 
-	public void fetchProjectsAndPackages() {
+	public DownloadResult<Void> fetchProjectsAndPackages() {
+		int confcheck = (baseUrl == null || baseUrl.isBlank()) ? 0 : 1;
+		confcheck += (token == null || token.isBlank()) ? 0 : 1;
+		if (confcheck == 0) {
+			return new DownloadResult<>(DownloadResult.Status.NO_CONFIG, "No Configuration"); //$NON-NLS-1$
+		}
+		if (confcheck == 1) {
+			return new DownloadResult<>(DownloadResult.Status.CONFIG_ERROR, "Configuration incomplete"); //$NON-NLS-1$
+		}
```

Comment:
To me the following would feel more readable:

```suggestion
		boolean missingUrl = baseUrl == null || baseUrl.isBlank();
		boolean missingToken = token == null || token.isBlank();
		if (missingUrl && missingToken) {
			return new DownloadResult<>(DownloadResult.Status.NO_CONFIG, "No Configuration"); //$NON-NLS-1$
		}
		if (missingUrl || missingToken) {
			return new DownloadResult<>(DownloadResult.Status.CONFIG_ERROR, "Configuration incomplete"); //$NON-NLS-1$
		}
```

@oberlehner Should the messages of ErrorMarkers be translateable?

Reply:
Yes, i would say so

Reply:
changed it to this version